### PR TITLE
add expression column type to UCR reports

### PR DESCRIPTION
**Pull request by czue**
this is the other thing I demo'd tuesday. the whole hierarchy of `Column` and `ColumnConfig` objects is super confusing in these reports and I would like to clean it up a bit more but decided to just put this up for now since I wasn't quite sure how to do that and there's messy dependencies on `SqlData`.

@emord might be a good person to review since he's been in this code recently. cc buddy @snopoke . probably easiest by commit.

### CODE REVIEW

File: corehq/apps/userreports/reports/specs.py [hunk 1]
```diff
@@ -24,11 +27,12 @@
     SimpleColumn,
     YearColumn,
 )
-from corehq.apps.reports.sqlreport import DatabaseColumn, AggregateColumn
+from corehq.apps.reports.sqlreport import DatabaseColumn, AggregateColumn, Column
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/specs.py]: F401 'Column' imported but unused

File: corehq/apps/userreports/sql/__init__.py [hunk 2]
```diff
@@ -1,3 +1,3 @@
 from .adapter import get_indicator_table, IndicatorSqlAdapter, metadata, ErrorRaisingIndicatorSqlAdapter
-from .columns import get_expanded_column_config, SqlColumnConfig
+from .columns import get_expanded_column_config, ColumnConfig
```
Comment (dimagimon) [inline corehq/apps/userreports/sql/__init__.py]: F401 'ColumnConfig' imported but unused
F401 'get_expanded_column_config' imported but unused

Comment (dimagimon) [inline corehq/apps/userreports/sql/__init__.py]: F401 'ColumnConfig' imported but unused
F401 'get_expanded_column_config' imported but unused

**Comment by emord** [comment]
kicking build

```
======================================================================
FAIL: test_report_runner2 (custom.ilsgateway.tests.test_report_runner.TestReportRunner)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/ilsgateway/tests/test_report_runner.py", line 166, in test_report_runner2
    ).first().total, 1
AssertionError: 0 != 1
```

File: corehq/apps/userreports/reports/specs.py [hunk 3]
```diff
@@ -333,6 +342,17 @@ class ExpressionColumn(BaseReportColumn):
     def wrapped_expression(self):
         return ExpressionFactory.from_spec(self.expression)
 
+    def get_column_config(self, data_source_config, lang):
+        return ColumnConfig(columns=[
+            CalculatedColumn(
+                header=self.get_header(lang),
+                slug=self.column_id,
+                # todo: are these needed?
+                # format_fn=self.get_format_fn(),
```
Comment (emord) [inline corehq/apps/userreports/reports/specs.py]: I think that you could have the `wrapped_expression` called in the `format_fn`. I think it gets called somewhere deep down in reports in `report_column.format_data()`. The exact syntax/structure you would need escapes me at the moment, but percent columns use this, to combine two columns which I think is what this is doing

**Review by emord — COMMENTED** [review]
Looks good once the tests pass. Agreed that refactoring this gets really messy because of the dependencies on reporting.

**Comment by czue** [comment]
:white_check_mark:

File: corehq/apps/userreports/reports/specs.py (hunk 3 above)
Comment (czue) [inline corehq/apps/userreports/reports/specs.py]: that's an interesting idea. I'll take a look at some point hopefully soonish

**Pull request merged**